### PR TITLE
Fix EthereumParser transaction type choice logic

### PR DESCRIPTION
**Pull request by Nana-EC**
Signed-off-by: Nana-EC <nana@swirldslabs.com>

**Description**:
Initially we had an optimization that observed the initial bytes of the raw transaction bytes when deciding the type of transaction parser.
Turns out the 2nd byte varies and so this optimization doesn't completely scale.
Additionally the RLP decoding is lazier than we thought and can be utilized to ensure the right transaction type parser is used.

- Update `CompositeEthereumTrnasactionParser` logic to improve parser choice of Legacy or Eip1559
- Add test case for issue transaction
- Cleaned up some `EthereumTransaction` related classes with orphaned fields and params

**Related issue(s)**:

Handles #3769 in 0.57

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/parser/record/ethereum/CompositeEthereumTransactionParserTest.java [hunk 1]
```diff
@@ -71,6 +74,13 @@ void decodeLessThanMinByteSize() {
                 .hasMessage(LESS_THAN_ERROR_MESSAGE);
     }
 
+    @SneakyThrows
+    @Test
+    void decodeEip1559WithAlternate2ndByte() {
+        var ethereumTransaction = ethereumTransactionParser.decode(Hex.decodeHex(LONDON_RAW_TX_2));
+        validateEthereumTransaction(ethereumTransaction);
```
Comment (steven-sheehy) [inline hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/parser/record/ethereum/CompositeEthereumTransactionParserTest.java:81]: Should assert type is specifically eip1559

File: hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/transactionhandler/EthereumTransactionHandler.java [hunk 2]
```diff
@@ -74,10 +74,10 @@ public void updateTransaction(Transaction transaction, RecordItem recordItem) {
         var contractFunctionResult = transactionRecord.hasContractCreateResult() ?
```
Comment (steven-sheehy) [inline hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/transactionhandler/EthereumTransactionHandler.java:74]: Should also cleanup every other local variable in this method then.

File: hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/ethereum/CompositeEthereumTransactionParser.java [hunk 3]
```diff
@@ -47,8 +48,27 @@ private EthereumTransactionParser getEthereumTransactionParser(byte[] transactio
             throw new InvalidDatasetException("Ethereum transaction bytes length is less than 2 bytes in length");
         }
 
-        var eip1559StartingBytesMatch = transactionBytes[0] == EIP1559_BYTES_PREFIX[0] &&
-                transactionBytes[1] == EIP1559_BYTES_PREFIX[1];
-        return eip1559StartingBytesMatch ? eip1559EthereumTransactionParser : legacyEthereumTransactionParser;
+        var decoder = RLPDecoder.RLP_STRICT.sequenceIterator(
+                transactionBytes);
```
Comment (steven-sheehy) [inline hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/ethereum/CompositeEthereumTransactionParser.java:52]: nit: Why is this on a separate line?

File: hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/ethereum/CompositeEthereumTransactionParser.java [hunk 4]
```diff
@@ -47,8 +48,27 @@ private EthereumTransactionParser getEthereumTransactionParser(byte[] transactio
...
+        if (legacyRlpItemByte != EIP1559_TYPE_BYTE) {
+            return legacyEthereumTransactionParser;
+        }
+
+        var eip1559RlpItem = decoder.next();
+        if (!eip1559RlpItem.isList()) {
+            return legacyEthereumTransactionParser;
+        }
+
+        return eip1559EthereumTransactionParser;
```
Comment (steven-sheehy) [inline hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/ethereum/CompositeEthereumTransactionParser.java:72]: We should use the exact same decision logic as [services](https://github.com/hashgraph/hedera-services/blob/a194837e3b7d755bde0de5c1130c86ddeb09cd2b/hapi-utils/src/main/java/com/hedera/services/ethereum/EthTxData.java#L83) uses:

```suggestion
        var decoder = RLPDecoder.RLP_STRICT.sequenceIterator(transactionBytes);
        var rlpItem = decoder.next();
        return rlpItem.isList() ? legacyEthereumTransactionParser : eip1559EthereumTransactionParser;
```

**Review by steven-sheehy — CHANGES_REQUESTED** [review]

File: hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/parser/record/ethereum/CompositeEthereumTransactionParserTest.java (hunk 1 above)
Comment (Nana-EC) [reply hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/parser/record/ethereum/CompositeEthereumTransactionParserTest.java:81]: Added

File: hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/transactionhandler/EthereumTransactionHandler.java (hunk 2 above)
Comment (Nana-EC) [reply hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/transactionhandler/EthereumTransactionHandler.java]: Updated

File: hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/ethereum/CompositeEthereumTransactionParser.java (hunk 3 above)
Comment (Nana-EC) [reply hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/ethereum/CompositeEthereumTransactionParser.java]: Realligned

File: hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/ethereum/CompositeEthereumTransactionParser.java (hunk 4 above)
Comment (Nana-EC) [reply hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/ethereum/CompositeEthereumTransactionParser.java]: Simplified

**Review by steven-sheehy — APPROVED** [review]
LGTM

**Pull request merged**